src/client/client.go

```diff
@@ -453,6 +453,34 @@ func NewForTest() (*APIClient, error) {
 	return client, nil
 }
 
+// NewEnterpriseClientForTest constructs a new APIClient for tests.
+func NewEnterpriseClientForTest() (*APIClient, error) {
```

Comment:
If this is only supposed to be used in tests, I like the pattern of depending on the interface `testing.TB` and then not returning an error and just using the require package to short-circuit on errors.

We use the naming scheme `NewTest____(t testing.TB) _____` elsewhere.

Reply:
Additionally you can register cleanup on the `testing.TB`

Reply:
Agree, this is an exact copy of `NewForTest` which has the same problems. I'm going to add a TODO, I think we could have much simpler implementations for both of them and move them to testutil.

Reply:
I don't think a package with test constructors is a good pattern generally. It produces dependency cycles.  If everything has a constructor in testutil then you can't import testutil for your unit test.  The testutil package currently has methods for generating random values, which has utility almost wherever you would write a test.  Something like serviceenv does make sense for black box testing, but my point is that constructors like this can't only live in a single package.

I didn't realize this was part of the public API, that is more of a problem.  I am generally fine with marking things as TODO, but this TODO would have to be addressed before 2.0, if we are serious about API compatibility. What is stopping this from just being moved into where the tests are?

`getUserMachineAddrAndOpt` <- Is that a better method to make public? Would users get value from that? 
Right now it seems like we are exporting 2 methods with no real value to users, which are just waiting to be depended on in a breaking way.

Reply:
The reason I copied the existing pattern and marked this TODO is to avoid having this discussion hold up this PR - I think we have technical debt around testing and the structure of the codebase, but my goal here is to try and land as much of the 2.0 feature work as possible so we can get feedback. I agree we should get this out of the public API before 2.0.

Reply:
Sure, as long as we are on the same page about the path forward, this doesn't need to be a blocker right now.